src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,61 @@
+package tasks;
+import java.io.Serializable;
+import java.time.LocalDate;
+/**
+ * Representation of a task
+ * recorded by the chatbot.
+ * @author Alvis Ng (supermii2)
+ */
+public abstract class Task implements Serializable {
+    /** Name of task */
+    protected String taskName;
+    /** One letter abbreviation for the task. */
+    protected String oneLetterAbbrev;
+    /** Whether the task has been completed */
+    private boolean completed;
+    /**
+     * Creates a task with the given name
+     * @param taskName Name of task
+     */
+    public Task(String taskName) {
+        this.taskName = taskName;
+        completed = false;
+    }
+    /**
+     * Checks if a task is completed
+     * @return State of completeness of task
+     */
+    public boolean isCompleted() {
+        return this.completed;
+    }
+    /**
+     * Sets the completion state of task
+     * to the given boolean
+     * @param completed State to change completion state
+     *     of task to.
+     */
+    public void setCompleted(boolean completed) {
+        if (this.completed == completed) {
+            if (completed) {
+                throw new IllegalArgumentException("Task is already marked!");
+            } else {
+                throw new IllegalArgumentException("Task is already unmarked");
+            }
+        }
+        this.completed = completed;
+    }
+    public String getName() {
+        return this.taskName;
+    }
+    public abstract LocalDate getDate();
+    /**
+     * String representation of Task
+     * @return String representation of task
+     */
+    @Override
+    public String toString() {
+        return this.isCompleted()
+            ? "[" + this.oneLetterAbbrev + "][X] " + this.taskName
+            : "[" + this.oneLetterAbbrev + "][ ] " + this.taskName;
+    }
```

Comment:
Just a suggestion to consider using String.format so the string's format is easier to see at first glance!